cpp/harvest/nacjd_data_publication.cc

```diff
@@ -0,0 +1,276 @@
+        }
+        if (not nacjd_doc.publisher_.empty()) {
+            publishing_260_info.emplace_back(MARC::Subfield('b', nacjd_doc.publisher_));
+        }
+        if (not nacjd_doc.year_pub_.empty()) {
+            publishing_info.emplace_back(MARC::Subfield('g', "year:" + nacjd_doc.year_pub_));
+            publishing_260_info.emplace_back(MARC::Subfield('c', nacjd_doc.year_pub_));
+        }
+        if (not nacjd_doc.volume_.empty()) {
+            publishing_info.emplace_back(MARC::Subfield('g', "volume:" + nacjd_doc.volume_));
+        }
+        if (not nacjd_doc.i_number_.empty()) {
+            publishing_info.emplace_back(MARC::Subfield('g', "number:" + nacjd_doc.i_number_));
+        }
+        if (not nacjd_doc.page_start_.empty() && not nacjd_doc.page_end_.empty()) {
+            publishing_info.emplace_back(MARC::Subfield('g', "pages:" + nacjd_doc.page_start_ + "-" + nacjd_doc.page_end_));
+        }
+
+        // construct url_info
+        std::vector<MARC::Subfield> urls;
+        if (not nacjd_doc.url_.empty()) {
+            urls.emplace_back(MARC::Subfield('a', nacjd_doc.url_));
+        }
+        if (not nacjd_doc.url_pdf_.empty()) {
+            urls.emplace_back(MARC::Subfield('a', nacjd_doc.url_pdf_));
+        }
+        if (not nacjd_doc.url_abs_.empty()) {
+            urls.emplace_back(MARC::Subfield('a', nacjd_doc.url_abs_));
+        }
+
+        // create a new record
+        MARC::Record new_record(MARC::Record::TypeOfRecord::LANGUAGE_MATERIAL, MARC::Record::BibliographicLevel::MONOGRAPH_OR_ITEM,
+                                "ICPSR" + nacjd_doc.ref_id_);
+
+
+        new_record.insertControlField("007", "cr||||");
+
+        if (not nacjd_doc.issn_.empty()) {
+            new_record.insertField("022", { { 'a', nacjd_doc.issn_ } });
+        }
+
+        if (not nacjd_doc.doi_.empty()) {
+            new_record.insertField("024", { { 'a', nacjd_doc.doi_ }, { '2', "doi" } }, '7');
+            new_record.insertField("856", { { 'u', "https://doi.org/" + nacjd_doc.doi_ }, { 'x', "R" }, { 'z', "LF" } },
+                                   /*indicator1 = */ '4',
+                                   /*indicator2 = */ '0');
+        }
+
+        if (not publishing_260_info.empty()) {
+            new_record.insertField("260", publishing_260_info, ' ', ' ');
+        }
+
+        new_record.insertField("041", { { 'a', "eng" } });
+        new_record.insertField("084", { { 'a', "2,1" }, { '2', "ssgn" } });
+        if (not authors.empty()) {
+            new_record.insertField("100", authors, '1');
+        }
+        if (not nacjd_doc.title_.empty()) {
+            new_record.insertField("245", { { 'a', nacjd_doc.title_ } }, '1', '0');
+        } else if (not nacjd_doc.sec_title_.empty()) {
+            new_record.insertField("245", { { 'a', nacjd_doc.sec_title_ } }, '1', '0');
+        }
+
+        if (not nacjd_doc.year_pub_.empty()) {
```

Comment:
Are the publisher and place of publication really unknown only because there is no year?